third_party/amd/lib/TritonAMDGPUToLLVM/ReduceOpToLLVM.cpp

```diff
@@ -17,6 +17,129 @@ using ::mlir::triton::gpu::getTotalElemsPerThread;
 
 namespace AMD {
 namespace {
+struct ReduceOpPromotionConversion
+    : public ConvertTritonGPUReduceScanToLLVMPattern<triton::ReduceOp> {
+public:
+  ReduceOpPromotionConversion(
+      TritonGPUToLLVMTypeConverter &typeConverter, ModuleAllocation &allocation,
+      ConvertTritonGPUOpToLLVMPatternBase::IndexCacheInfo &indexCacheInfo,
+      int computeCapability, PatternBenefit benefit)
+      : ConvertTritonGPUReduceScanToLLVMPattern<triton::ReduceOp>(
+            typeConverter, allocation, indexCacheInfo, benefit),
+        computeCapability(computeCapability) {}
+
+  LogicalResult
+  matchAndRewrite(triton::ReduceOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    
+    // promote op
+    rewriter.modifyOpInPlace(op, [&]() {
+      // promote operands
+      SmallVector<Value> newOperands;
+      for (OpOperand &operand : op->getOpOperands()) {
```

Comment:
It doesn't look like using a conversion pattern is the right thing here. Also you are using op->getOperands() which might have already gone through conversion at this time.
I would suggest make it a RewritePattern and running it as a pre-processing. This would also save you from overriding the `benefit`

Reply:
I will look into this.

Reply:
moved to preprocessing step